lib/ruby_lsp/encoder/relative.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module RubyLsp
+  module Encoder
+    class Relative
+      TOKEN_TYPES = [
+        :variable,
+        :method,
+      ].freeze
+      TOKEN_MODIFIERS = [].freeze
+
+      def initialize(request)
```

Comment:
It feels a bit odd to me that the encoder runs the request. Is there an advantage to this approach rather than receiving the array of tokens?